src/snowflake/connector/query_context_cache.py

```diff
@@ -180,91 +169,115 @@ def _get_elements(self) -> set[QueryContextElement]:
-
-            if data is None or len(data) == 0:
-                self.clear_cache()
-                logger.debug("deserialize_from_arrow_base64() returns")
-                self.log_cache_entries()
-                return
-
-            decoded_data = b64decode(data)
-            input = BytesIO(decoded_data)
-            try:
-                with pa.ipc.open_stream(input) as reader:
-                    for record_batch in reader:
-                        record_dict = record_batch.to_pydict()
-                        for i in range(len(record_batch)):
-                            self.merge(
-                                record_dict["id"][i],
-                                record_dict["timestamp"][i],
-                                record_dict["priority"][i],
-                                record_dict["context"][i],
-                            )
-            except Exception as e:
-                logger.debug(f"deserialize_from_arrow_base64: Exception = {e}")
-                # clear cache due to incomplete merge
-                self.clear_cache()
-
-            self.check_cache_capacity()
-            logger.debug("deserialize_from_arrow_base64() returns")
-            self.log_cache_entries()
-
-    def serialize_to_arrow_base64(self) -> str:
-        if not installed_pandas:
-            return self._data
-
+    
+    def serialize_to_json(self) -> str:
         with self._lock:
-            logger.debug("serialize_to_arrow_base64() called")
+            logger.debug("serialize_to_json() called")
             self.log_cache_entries()
 
             if len(self._tree_set) == 0:
                 return None
 
             try:
-                stream = BytesIO()
                 size = self.get_size()
                 id_vals = [None] * size
                 timestamp_vals = [None] * size
                 priority_vals = [None] * size
                 context_vals = [None] * size
                 self.get_elements(id_vals, timestamp_vals, priority_vals, context_vals)
-
-                with pa.ipc.RecordBatchStreamWriter(
-                    stream, QUERY_CONTEXT_SCHEMA
-                ) as writer:
-                    id_array = pa.array(id_vals, type=pa.int64())
-                    timestamp_array = pa.array(timestamp_vals, type=pa.int64())
-                    priority_array = pa.array(priority_vals, type=pa.int64())
-                    context_array = pa.array(context_vals, type=pa.binary())
-                    record_batch = pa.record_batch(
-                        [id_array, timestamp_array, priority_array, context_array],
-                        schema=QUERY_CONTEXT_SCHEMA,
-                    )
-                    writer.write_batch(record_batch)
-
-                stream.seek(0)
-                # use same encoding use on jdbc driver
-                data = b64encode(stream.read()).decode("iso-8859-1")
-                stream.close()
+                data = {}
+                if(size >= 1):
```

Comment:
```suggestion
                if size > 0:
```
a more pythonic way or writing

Reply:
updated